packages/stencil/src/components/datacom-radio/datacom-radio.tsx

```diff
@@ -98,86 +145,60 @@ export class DatacomRadio implements FormControl {
-                formenctype={this.formenctype}
-                formnovalidate={this.formnovalidate}
-                name={this.name}
-                type={this.type}
-                checked={this.checked}
-                disabled={this.disabled}
-                required={this.required}
-                value={this.value}
-                id={this.inputId}
-                class={classes}
-                onChange={this.handleChange}
-              />
-
-              <label htmlfor={this.inputId} class="dc-radio-label">
-                <span class={`dc-radio-image-${this.imagePosition}`}>
-                  {image}
-                  {this.label}
-
-                  <slot></slot>
-                </span>
-              </label>
-            </div>
-          </Host>
-        );
-      } else {
+
+      {
         return (
           <Host>
-            <div class="dc-radio-wrapper">
-              <input
-                form={this.form}
-                autofocus={this.autofocus}
-                formmethod={this.formmethod}
-                formaction={this.formaction}
-                formtarget={this.formtarget}
-                formenctype={this.formenctype}
-                formnovalidate={this.formnovalidate}
-                name={this.name}
-                type={this.type}
-                checked={this.checked}
-                disabled={this.disabled}
-                required={this.required}
-                value={this.value}
-                id={this.inputId}
-                class={classes}
-                onChange={this.handleChange}
-              />
-              <label htmlFor={this.inputId} class="dc-radio-label">
-                <span class={`dc-radio-image-${this.imagePosition}`}>
-                  {this.label}
-                  <slot></slot>
-                </span>
-              </label>
+            <div class={classes}>
+              <div class="dc-radio-wrapper">
+                <input
+                  class="dc-radio-input"
+                  ref={el => this.setInputElementRef(el)}
+                  form={this.form}
+                  autofocus={this.autofocus}
+                  formmethod={this.formmethod}
+                  formaction={this.formaction}
+                  formtarget={this.formtarget}
+                  formenctype={this.formenctype}
+                  formnovalidate={this.formnovalidate}
+                  name={this.name}
+                  type={this.type}
+                  checked={this.checked}
+                  disabled={this.disabled}
+                  required={this.required}
+                  value={this.value}
+                  id={this.inputId}
+                  onChange={this.handleChange}
+                  tabIndex={0}
+                />
+                <label tabIndex={0} htmlFor={this.inputId} class="dc-radio-label">
+                  <span tabIndex={-1} class={`dc-radio-image-${this.imagePosition}`}>
+                    {image}
```

Comment:
Didn't you need to conditionally remove this image depending on if it's a 'radio' or 'button' variant?

Reply:
Fixed it.